test/spec/Feature/Query/UpdateSpec.hs

```diff
@@ -388,6 +388,17 @@ spec = do
           }
 
   context "limited update" $ do
+    it "does not work when no limit query or filter is given" $
+      request methodPatch "/limited_update_items"
+          [("Prefer", "tx=commit"), ("Prefer", "count=exact")]
+          [json| {"name": "updated-item"} |]
+        `shouldRespondWith`
+          ""
+          { matchStatus  = 404
+          , matchHeaders = [ matchHeaderAbsent hContentType
+                           , "Content-Range" <:> "*/0" ]
+          }
+
```

Comment:
This error was just a lucky side-effect of https://github.com/PostgREST/postgrest/issues/2343. This needs a clear error message plus a 422 status - like done for [batch delete](https://github.com/PostgREST/postgrest/pull/2355#issuecomment-1176973562).